_posts/2018/2018-02-18-avoid-writing-sql-when-using-activerelation.md

```diff
@@ -0,0 +1,74 @@
+---
+title: 'Avoid Writing SQL When Using ActiveRelation'
+description: 'It's fine, but prefer not to if you can.'
+layout: article
+category: ruby
+image:
+  base: '2018/avoid-writing-sql-when-using-activerelation'
+  alt: 'Coding SQL'
+  source: 'https://unsplash.com/photos/fPkvU7RDmCo'
+  credit: 'Caspar Rubin'
+---
+
+ActiveRelation, which powers the searching and querying engine of ActiveRecord, is a powerful and flexible tool.
+
+
+## Instead of...
+
+...writing literal SQL strings inside an ActiveRelation `#where` method.
+
+```ruby
+Person.where("name = #{ params[:name] } and hidden_at is null")
+```
+
+
+## Or...
+
+...writing literal strings and using the Array method to sanitise data.
+
+```ruby
```

Comment:
So because of where `params[:name]` is interpolated in in the previous example, the data won't be sanitised?

If so, is it really the Array method doing the sanitation or something else..? Because both use arrays...

Reply:
(..it's more the presence of string interpolation, right?)

Reply:
The interpolation version doesn't sanitise. The 'array' version does.

Reply:
Yes, so I think you should stress that it's the presence of interpolation that stops sanitisation...because an array is used in both cases.

Reply:
You mean a `String` right?